### PR TITLE
modify logic for mediaframework

### PR DESCRIPTION
**Pull request by jsdosa**
1. fix compile error of media player
2. add namespace Media for media recorder
3. modify example

Signed-off-by: jaesick.shin <jaesick.shin@samsung.com>

### CODE REVIEW

File: apps/examples/cxxtest/MediaRecorder.hpp [hunk 1]
```diff
@@ -10,68 +10,71 @@
...
-	template<typename _Callable, typename... _Args>
-	void enqueue(_Callable&& __f, _Args&&... __args)
-	{
-		unique_lock<std::mutex> lock(*qMtx);
-		std::function<void()> func = std::bind(std::forward<_Callable>(__f), std::forward<_Args>(__args)...);
-		cmdQueue.push(func);
-		cvQueue.notify_one();
-	}
+		template<typename _Callable, typename... _Args>
+		void enqueue(_Callable&& __f, _Args&&... __args)
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.hpp]: Please do not use void, user should check return type of binded function..

File: apps/examples/cxxtest/MediaRecorder.hpp [hunk 2]
```diff
@@ -10,68 +10,71 @@
...
-	recorder_result_t pause();
-	recorder_result_t stop();
+		recorder_result_t start();
+		recorder_result_t pause();
+		recorder_result_t stop();
 
-	recorder_result_t getVolume() const;
-	void setVolume(int vol);
+		recorder_result_t getVolume() const;
+		void setVolume(int vol);
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.hpp]: why void?

File: apps/examples/cxxtest/MediaRecorder.hpp [hunk 3]
```diff
@@ -10,68 +10,71 @@
...
 
-	void setDataSource(DataSource dataSource);
+		void setDataSource(DataSource dataSource);
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.hpp]: same as above

File: apps/examples/cxxtest/MediaRecorder.hpp [hunk 4]
```diff
@@ -10,68 +10,71 @@
...
-	recorder_result_t destroy();
+		recorder_result_t create();
+		recorder_result_t destroy();
 
-	recorder_result_t start();
-	recorder_result_t pause();
-	recorder_result_t stop();
+		recorder_result_t start();
+		recorder_result_t pause();
+		recorder_result_t stop();
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.hpp]: prepare needed.

File: apps/examples/cxxtest/MediaRecorder.hpp [hunk 5]
```diff
@@ -10,68 +10,71 @@
...
-	~MediaRecorder();
+	class MediaRecorder
+	{
+	public:
+		MediaRecorder();
+		~MediaRecorder();
 
-	recorder_result_t create();
-	recorder_result_t destroy();
+		recorder_result_t create();
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.hpp]: add some comment about below apis.
Below apis changes state, and each api should follow pre-state
for example, start should be called after prepare, stop should be called after strt or pause...

File: apps/examples/cxxtest/MediaRecorder.cxx [hunk 6]
```diff
@@ -1,133 +1,129 @@
...
 
-	return 0;
-}
+			run();
+		}
+
+		return 0;
+	}
 
+}
```
Comment (Taejun-Kwon) [inline apps/examples/cxxtest/MediaRecorder.cxx]: please check this.

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 1 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: we can not give any return type at the time of enque. Later, we need to define return type from internal function to callback.

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 3 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: yes i wil change that

File: apps/examples/cxxtest/MediaRecorder.cxx (hunk 6 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.cxx]: it is correct. that brace is namespace pair.

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 4 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: ok, i will add

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 5 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: The above is currently being implemented.

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 2 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: i will change that, next commit.
but, how to support related volume API in alsa layer?

File: apps/examples/cxxtest/MediaRecorder.hpp (hunk 3 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.hpp]: The DataSource has not been implemented yet and will be fixed later.

File: apps/examples/cxxtest/MediaRecorder.cxx (hunk 6 above)
Comment (jsdosa) [reply apps/examples/cxxtest/MediaRecorder.cxx]: i will modify that.

**Pull request merged**